blockchain_connector/avalon_blockchain_connector/fabric/fabric_connector.py

```diff
@@ -75,17 +76,43 @@ def sync_worker(self):
         """
         Check for existing worker and update worker to fabric blockchain
         """
-        worker_type = WorkerType.TEE_SGX
+        # First get all the workers from the blockchain
         req_id = 15
-        lookup_result = self.__jrpc_worker.worker_lookup(
+        bc_workers = self.__fabric_worker.worker_lookup(
+            worker_type=WorkerType.TEE_SGX,
+        )
+        logging.info("Workers in blockchain {}".format(
+            bc_workers
+        ))
+        # Set worker status to Decommissioned.
+        if bc_workers and bc_workers[0] > 0:
+            for worker in bc_workers[2]:
+                update_status = self.__fabric_worker.worker_set_status(
+                    worker,
+                    WorkerStatus.DECOMMISSIONED
+                )
+                if update_status == ContractResponse.SUCCESS:
+                    logging.info("Set worker {} status to {}".format(
+                        worker,
+                        WorkerStatus.DECOMMISSIONED
+                    ))
+                else:
+                    logging.info("Failed to set worker {} status".format(
+                        worker
+                    ))
+        else:
+            logging.info("No active workers in blockchain")
+        # Get active workers from avalon
+        worker_type = WorkerType.TEE_SGX
+        active_workers = self.__jrpc_worker.worker_lookup(
```

Comment:
The workerLookup in TC spec return all workers that matches all input parameters and worker status is not one of the them. We can add a new contract (out of spec) to implement workLookup_active or update the spec to include worker status as one of input parameter.  We need to make the logic of syncing workers between shared KV and block chain uniform. So this could be handled as part of it